<!DOCTYPE html>
<html lang="en"><head>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-html/tabby.min.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.2.335">

  <meta name="author" content="Karl Tayeb">
  <title>Karl’s Website - Generalizing SuSiE</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <link rel="stylesheet" href="../../site_libs/revealjs/dist/theme/quarto.css" id="theme">
  <link href="../../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-captioned.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-captioned) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-captioned.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-captioned .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-captioned .callout-caption  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-captioned.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-captioned.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-caption {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-caption {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-caption {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-captioned .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-captioned .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-captioned) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-caption {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-caption {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-caption {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-caption {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-caption {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Generalizing SuSiE</h1>
  <p class="subtitle">Stephens’ Group Meeting</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Karl Tayeb 
</div>
</div>
</div>

  <p class="date">1/23/24</p>
</section>
<section id="plan" class="slide level2">
<h2>Plan</h2>
<ul>
<li>Talk about our current approach, GIBSS</li>
<li>Talk about (generalized) additive models</li>
</ul>
</section>
<section id="ibss" class="slide level2">
<h2>IBSS</h2>

<img data-src="resources/wang18_ibss.png" style="width:40.0%" class="r-stretch quarto-figure-center"><p class="caption">Algorithm 1 from <span class="citation" data-cites="wangSimpleNewApproach">(<a href="#/references" role="doc-biblioref" onclick="">1</a>)</span></p></section>
<section id="generalized-ibss" class="slide level2">
<h2>Generalized IBSS</h2>
<ul>
<li>GLM <span class="math inline">\(\mu(x) = \mathbb E[Y |X = x]\)</span>. <span class="math inline">\(g(\mu(x)) = \sum_l \psi_l = \sum_l x^T \beta_l\)</span></li>
<li>Idea: plug in point estimates <span class="math inline">\({\hat\psi} _j, j\neq l\)</span> to estimate <span class="math inline">\(\psi_l\)</span></li>
</ul>

<img data-src="resources/gibss.png" style="width:40.0%" class="r-stretch"></section>
<section id="variational-objective" class="slide level2 smaller">
<h2>Variational Objective</h2>
<p>log joint:</p>
<p><span class="math display">\[\begin{align}
F(\beta_1, \dots, \beta_L) &amp;= \log p(y | \sum_l(\psi_l)) + \sum_{l=1}^L\log p(\psi_l), \\
\psi_l &amp;= X \beta_l, \\
\beta_l &amp;\sim g_l
\end{align}\]</span></p>
<p>ELBO:</p>
<p><span class="math display">\[\begin{align}
\mathcal F(q_1, \dots, q_L) &amp;= \mathbb E_q \left[ F(\beta_1, \dots \beta_L) \right] + \sum_l H(q_l) \\
q(\beta_1, \dots, \beta_L) &amp;= \prod q_l(\beta_l)
\end{align}\]</span></p>
</section>
<section id="gibss-as-a-first-order-approximation-to-cavi" class="slide level2 smaller">
<h2>GIBSS as a first order approximation to CAVI</h2>
<p>Exponential family <span class="math inline">\(f(y | \eta) \propto \exp \{y \eta - A(\eta) \}\)</span>, <span class="math inline">\(A\)</span> convex. Additive model: <span class="math inline">\(\eta = \sum \eta_l\)</span></p>
<p><span class="math display">\[\begin{align}
\mathbb E_q[\log f(y | \eta)]
&amp;= \mathbb E_{q_l} [\mathbb E_{q_{-l}}[\log f(y | \eta_l + \eta_{-l})] | \eta_l] \\
&amp;\leq \mathbb E_{q_l} [\mathbb E_{q_{-l}}[\log f(y | \eta_l + \bar\eta_{-l}) + \nabla \log f_{\eta_{-l}}(\bar \eta_{-l})) (\eta_{-l} - \bar \eta_{-l})] | \eta_l] \\
&amp;= \mathbb E_{q_l} [\log f(y | \eta_l + \bar\eta_{-l})] \\
\end{align}\]</span></p>
<p>Notes:</p>
<ul>
<li>For exponential families, we <em>upper bound</em> the ELBO for each coordinate update, a bit awkward.</li>
<li>We use a different approximation at each step, not clear if the algorithm converges.</li>
</ul>
</section>
<section id="generalized-ser" class="slide level2">
<h2>Generalized SER</h2>

<img data-src="resources/gser.png" style="width:40.0%" class="r-stretch"></section>
<section id="approximating-generalized-ser" class="slide level2">
<h2>Approximating Generalized SER</h2>

<img data-src="resources/gsermle.png" style="width:40.0%" class="r-stretch"></section>
<section id="additive-effects-model" class="slide level2">
<h2>Additive effects model</h2>
<p><span class="math display">\[\begin{align}
{\bf y} | \mu &amp; \sim N(\mu, \sigma^2) \\
\mu &amp;= \sum_l \mu_l \\
\mu_l &amp;\sim g_l
\end{align}\]</span></p>
<p>SuSiE is the special case where <span class="math inline">\(g_l \sim SER\)</span></p>
</section>
<section id="backfitting-in-the-additive-regression-model" class="slide level2">
<h2>Backfitting in the additive regression model</h2>

<img data-src="resources/hastie86_backfitting.png" style="width:40.0%" class="r-stretch quarto-figure-center"><p class="caption"><span class="citation" data-cites="hastieGeneralizedAdditiveModels1986">(<a href="#/references" role="doc-biblioref" onclick="">2</a>)</span></p></section>
<section id="rationale-for-maximum-likelihood-estimation" class="slide level2">
<h2>Rationale for maximum likelihood estimation</h2>

<img data-src="resources/hastie87deviance.png" style="width:40.0%" class="r-stretch quarto-figure-center"><p class="caption">Fig 1. reproduced from <span class="citation" data-cites="hastieCloserLookDeviance1987">(<a href="#/references" role="doc-biblioref" onclick="">3</a>)</span></p></section>
<section id="additive-effects-model-1" class="slide level2">
<h2>Additive effects model</h2>

<img data-src="resources/wang18_additive_effects.png" style="width:40.0%" class="r-stretch quarto-figure-center"><p class="caption"><span class="citation" data-cites="wangSimpleNewApproach">(<a href="#/references" role="doc-biblioref" onclick="">1</a>)</span></p></section>
<section id="generalized-additive-model" class="slide level2">
<h2>Generalized additive model</h2>
<p><span class="math inline">\(g\)</span> the canonical link function, <span class="math inline">\(\mathbb E[y] = g^{-1}(\eta)\)</span></p>
<p><span class="math display">\[\begin{align}
y | \eta &amp;\sim p_{g^{-1}(\eta)} \\
\eta &amp;= \sum_l \eta_l \\
\eta_l &amp;\sim g_l
\end{align}\]</span></p>
</section>
<section id="generalized-local-scoring" class="slide level2">
<h2>Generalized local scoring</h2>

<img data-src="resources/hastie86_generalized_local_scoring.png" style="width:40.0%" class="r-stretch quarto-figure-center"><p class="caption"><span class="citation" data-cites="hastieGeneralizedAdditiveModels1986">(<a href="#/references" role="doc-biblioref" onclick="">2</a>)</span></p></section>
<section id="comparison-to-gibss" class="slide level2">
<h2>Comparison to GIBSS</h2>
<ul>
<li>Generalized local scoring iteratively updates weights, and fits the weighted additive model</li>
<li>GIBSS performs IRLS for each variable, fixing <span class="math inline">\(L-1\)</span> additive components</li>
<li>Estimates don’t</li>
</ul>
</section>
<section id="variational-objective-1" class="slide level2 smaller">
<h2>Variational Objective</h2>
<p>log joint:</p>
<p><span class="math display">\[\begin{align}
F(\beta_1, \dots, \beta_L) &amp;= \log p(y | \sum_l(\psi_l)) + \sum_{l=1}^L\log p(\psi_l), \\
\psi_l &amp;= X \beta_l, \\
\beta_l &amp;\sim g_l
\end{align}\]</span></p>
<p>ELBO:</p>
<p><span class="math display">\[\begin{align}
\mathcal F(q_1, \dots, q_L) &amp;= \mathbb E_q \left[ F(\beta_1, \dots \beta_L) \right] + \sum_l H(q_l) \\
q(\beta_1, \dots, \beta_L) &amp;= \prod q_l(\beta_l)
\end{align}\]</span></p>
</section>
<section id="mode-rather-than-mean" class="slide level2 smaller">
<h2>Mode rather than mean?</h2>
<p><span class="math display">\[\begin{align}
F(\beta_1, \dots, \beta_L) &amp;= \log p(y | \sum_l(\psi_l)) + \sum_{l=1}^L\log p(\psi_l), \\
\end{align}\]</span></p>
<p><span class="math display">\[\begin{align}
\beta_l^* = \arg\max_{\beta_l}  F(\beta_1, \dots, \beta_L)
\end{align}\]</span></p>
<p>Notes:</p>
<ul>
<li>Converges to a mode of the posterior.</li>
<li>Just iterative stepwise regression <em>with fixed <span class="math inline">\(L\)</span></em>.</li>
<li>Seems undesirable to include <span class="math inline">\(\beta_l\)</span> for effects with weak evidence.</li>
<li>Resembles CAVI when <span class="math inline">\(q(\gamma_l)\)</span> is concentrated on highly correlated variables.</li>
<li>Can still report SER posteriors for each effect <span class="math inline">\(l\)</span>.</li>
</ul>
</section>
<section id="laplace-vs-wakefield-review" class="slide level2">
<h2>Laplace vs Wakefield (Review)</h2>

<img src="https://karltayeb.github.io/research/logistic_abf_accuracy/index_files/figure-html/unnamed-chunk-4-1.png" class="r-stretch quarto-figure-center"><p class="caption">Figure&nbsp;1: Wakefield’s ABF can be order of magnitude off when the <span class="math inline">\(z\)</span>-score is large</p></section>
<section id="problems-with-wakefield-review" class="slide level2">
<h2>Problems with Wakefield (Review)</h2>
<p><span class="math display">\[
\text{ABF} = \sqrt{\frac{V+W}{V}} \exp (- \frac{z^2}{2} \frac{W}{V + W})
\]</span></p>

<img data-src="resources/abf_biased.png" class="r-stretch"></section>
<section id="susie-rss-and-the-wakefield-bf" class="slide level2">
<h2>SuSiE-RSS and the Wakefield BF</h2>
<ol type="1">
<li>Recall that Wakefield’s ABF is not accurate when the <span class="math inline">\(z\)</span>-score is large</li>
<li>Applying SuSiE-RSS to summary statistics from some other model besides Gaussian linear model</li>
</ol>
</section>
<section id="how-large-do-the-z-scores-need-to-be" class="slide level2">
<h2>How large do the <span class="math inline">\(z\)</span>-scores need to be?</h2>

<img data-src="resources/ser_fdp_by_max_z.png" class="r-stretch"></section>
<section>
<section id="end" class="title-slide slide level1 center">
<h1>End</h1>

</section>
<section id="a" class="slide level2">
<h2>A</h2>
<p><span class="math display">\[\begin{align}
\mathcal F(q_1, \dots, q_L) = \mathbb E_q \left[ F(\beta_1, \dots \beta_L) \right] + \sum_l H(q_l)
\end{align}\]</span></p>
<p>Coordinate ascent on <span class="math inline">\((\beta_l)\)</span>. For SuSiE, at each step fit an SER, but only return the posterior mode of the SER, rather than the mean. Essentially stepwise selection, but reports a posterior distribution for each effect. ## Backfitting</p>
<p>Maximize the <em>expected</em> log likelihood</p>
<p><span class="math display">\[
\begin{align}
\mathbb E[ l(\hat{\eta}(X), Y)] = \max_{\eta} \mathbb E[l(\eta(X), Y)]
\end{align}
\]</span></p>
</section>
<section id="overview" class="slide level2">
<h2>Overview</h2>
<ul>
<li>Publication plan:
<ul>
<li>Generalized SuSiE via IBSS, emphasis on fine-mapping for non-Gaussian models</li>
<li>GSEA with logistic SuSiE</li>
</ul></li>
</ul>
</section>
<section id="fine-mapping-under-the-multivariate-gaussian-model" class="slide level2">
<h2>Fine-mapping under the multivariate Gaussian model</h2>
<p>Most fine-mapping methods assume summary statistics from marginal association studies are normally distributed, with covariance determined by LD <sup>1</sup></p>
<p><span class="math display">\[\begin{align*}
\hat{ {\bf z} }  \sim N({\bf z}, R)
\end{align*}\]</span></p>
<p>Statistical property of OLS– what if the marginal effects are coming from somewhere else?</p>
<aside><ol class="aside-footnotes"><li id="fn1"><p>see <span class="citation" data-cites="zhurss">(<a href="#/references" role="doc-biblioref" onclick="">4</a>)</span>, <span class="citation" data-cites="lozanoMultivariateNormalDistribution2017">(<a href="#/references" role="doc-biblioref" onclick="">5</a>)</span>, <span class="citation" data-cites="hormozdiariIdentifyingCausalVariants2014">(<a href="#/references" role="doc-biblioref" onclick="">6</a>)</span>, <span class="citation" data-cites="susierss2022">(<a href="#/references" role="doc-biblioref" onclick="">7</a>)</span></p></li></ol></aside></section>
<section id="what-do-we-want-to-accomplish" class="slide level2">
<h2>What do we want to accomplish?</h2>
<ol type="1">
<li>Establish <em>when</em> there is a problem with fine-mapping with summary stats from non-Gaussian models</li>
<li>(Hopefully) find that these situations are not uncommon</li>
<li>(Hopefully) demonstrate that GIBSS offers improvement in these situations</li>
<li>(Fallback) advise people to fine map with summary statistics from linear models</li>
</ol>
</section>
<section id="three-horse-race" class="slide level2 smaller">
<h2>Three horse race</h2>
<table>
<colgroup>
<col style="width: 33%">
<col style="width: 46%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Method</strong></th>
<th><strong>Notes</strong></th>
<th><strong>Summary stats</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Generalized IBSS</td>
<td>“correct” model, hueristic algorithm</td>
<td>No</td>
</tr>
<tr class="even">
<td><em>Logistic + RSS</em></td>
<td>ad-hoc, actually used <span class="citation" data-cites="wightmanGenomewideAssociationStudy2021 kurkiFinnGenProvidesGenetic2023">(<a href="#/references" role="doc-biblioref" onclick="">8</a>,<a href="#/references" role="doc-biblioref" onclick="">9</a>)</span></td>
<td>Yes</td>
</tr>
<tr class="odd">
<td>Linear + RSS</td>
<td>mis-specified model, correct algorithm</td>
<td>Yes</td>
</tr>
</tbody>
</table>
</section>
<section id="gibss-overview" class="slide level2">
<h2>GIBSS overview</h2>
<ol type="1">
<li>Compute univariate effect estimates using regression of choice, must return MLE and stderr</li>
<li>Compute/approximate BFs and posterior means: Laplace, quadrature, etc.</li>
<li>Use predictions as fixed offsets when updating next effect</li>
<li>Iterate until <del>convergence</del> (we don’t know if it converges)</li>
</ol>
</section>
<section id="key-questions" class="slide level2 scrollable">
<h2>Key questions</h2>
<ul>
<li>Generalized IBSS vs Logistic RSS
<ul>
<li><span class="math inline">\(L=1\)</span> case reduces to IBSS-Laplace vs IBSS-ABF</li>
<li>GIBSS-Laplace should be strictly better, but by how much, and when?</li>
<li><span class="math inline">\(L &gt; 1\)</span> all bets are off with Logistic RSS</li>
</ul></li>
<li>Linear SuSiE vs Logistic RSS
<ul>
<li>Helpful to look at <span class="math inline">\(L=1\)</span> case, Linear + Wakefield vs Logistic + Wakefield</li>
<li>In GWAS linear regression is often a good approximation to logistic regression</li>
</ul></li>
<li>Linear SuSiE vs Generalized IBSS
<ul>
<li>When does linear SuSiE give reliable results?</li>
<li>When does GIBSS provide and advantage?</li>
</ul></li>
</ul>
</section>
<section id="potential-problems" class="slide level2">
<h2>Potential problems</h2>
<p><strong>Logistic + RSS</strong></p>
<ul>
<li>Covariate of marginal <span class="math inline">\(z\)</span>-scores do <strong>not</strong> correspond with LD</li>
<li>Under appreciated source of “LD Mismatch?”</li>
<li>Follow-up: what is the correct covariance matrix?</li>
<li>Inherits problems from using ABF– not the most accurate</li>
</ul>
<p><strong>Logistic GIBSS</strong></p>
<ul>
<li>Marginal effect estimates are biased when there is a large genetic component</li>
</ul>
</section>
<section id="correlation-of-marginal-effects-in-logistic-regression" class="slide level2">
<h2>Correlation of marginal effects in logistic regression:</h2>
<ol type="1">
<li>Simulate <span class="math inline">\(\begin{bmatrix}x_1 \\ x_2 \end{bmatrix} \sim N(\begin{bmatrix}0 \\ 0 \end{bmatrix}, \begin{bmatrix}1 &amp; \rho \\ \rho &amp; 1 \end{bmatrix})\)</span></li>
<li>Simulate <span class="math inline">\(y\)</span> under a logistic model <span class="math inline">\(y \sim Bin(1, \sigma(\psi)), \; \psi := b_0 + b x_1\)</span></li>
<li><span class="math inline">\(cor(z_1, z_2) \neq \rho \implies\)</span> LD matrix is the wrong covariance matrix</li>
</ol>
</section>
<section id="correlation-of-marginal-effects-cont" class="slide level2 nostretch">
<h2>Correlation of marginal effects (cont)</h2>
<p><strong>An under-appreciated source of “LD mismatch”?</strong></p>
<p><span class="math inline">\(n = 500\)</span>, <span class="math inline">\(b_0 = -1\)</span>, <span class="math inline">\(b = 0, 1, 2, 3\)</span> <sup>1</sup></p>
<p><img data-src="resources/logistic_marginal_z_cor.png" style="width:40.0%"></p>
<aside><ol class="aside-footnotes"><li id="fn2"><p><a href="https://karltayeb.github.io/research/logistic_regression_marginal_effects_exploration/#demonstrations">more simulations settings here</a></p></li></ol></aside></section>
<section id="laplace-vs-wakefield-review-1" class="slide level2">
<h2>Laplace vs Wakefield (Review)</h2>

<img src="https://karltayeb.github.io/research/logistic_abf_accuracy/index_files/figure-html/unnamed-chunk-4-1.png" class="r-stretch quarto-figure-center"><p class="caption">Figure&nbsp;2: Wakefield’s ABF can be order of magnitude off when the <span class="math inline">\(z\)</span>-score is large</p></section>
<section id="problems-with-wakefield-review-1" class="slide level2">
<h2>Problems with Wakefield (Review)</h2>
<p>!()[resources/abf_biased.png]</p>
<p>!()[resources/abf_eq.png]</p>
</section>
<section id="susie-rss-and-the-wakefield-bf-1" class="slide level2">
<h2>SuSiE-RSS and the Wakefield BF</h2>
<ol type="1">
<li>Recall that Wakefield’s ABF is not accurate when the <span class="math inline">\(z\)</span>-score is large</li>
<li>Applying SuSiE-RSS to summary statistics from some other model besides Gaussian linear model</li>
</ol>
</section>
<section id="how-large-do-the-z-scores-need-to-be-1" class="slide level2">
<h2>How large do the <span class="math inline">\(z\)</span>-scores need to be?</h2>

<img data-src="resources/ser_fdp_by_max_z.png" class="r-stretch"></section>
<section id="biased-effect-estimates-and-bfs" class="slide level2">
<h2>Biased effect estimates (and BFs)</h2>
<p>Simulation: one causal variant in the locus that explains <span class="math inline">\(1\%\)</span> of heritability of liability. <span class="math inline">\(h^2 = 0.1, 0.2, 0.5, 0.9\)</span></p>
<p><span class="math display">\[\begin{align*}
y \sim Bin(1, \sigma(\psi)) \\
\psi = b_0 + b x + \epsilon \\
\epsilon \sim N(0, \sigma^2)
\end{align*}\]</span></p>
</section>
<section id="biased-effect-estimates-and-bfs-1" class="slide level2">
<h2>Biased effect estimates (and BFs)</h2>

<img data-src="resources/random_intercept_sim_table.png" class="r-stretch"></section>
<section id="biased-effect-estimates-and-bfs-2" class="slide level2">
<h2>Biased effect estimates (and BFs)</h2>

<img data-src="resources/random_intercept_sim_plot.png" class="r-stretch quarto-figure-center"><p class="caption">95% C.I. for different <span class="math inline">\(h^2\)</span></p></section>
<section id="biased-effect-estimates-and-bfs-3" class="slide level2">
<h2>Biased effect estimates (and BFs)</h2>
<ul>
<li>For phenotypes with substantial <span class="math inline">\(h^2\)</span> of liability, restricting our attention to a single locus will lead to biased effect estimates</li>
<li>Remark: linear model doesn’t struggle with this issue because in practice we estimate the residual variance (or set conservatively)</li>
<li>Remark:basically a random intercept model, but this seems a little different than the usual motivation for mixed model approaches.</li>
</ul>
</section>
<section id="real-data-analsis" class="slide level2">
<h2>Real data analsis</h2>
<ul>
<li><strong>Q: if we reun SuSiE, GIBSS, Logistic + RSS on real case-control GWAS do we get qualitatively different results?</strong></li>
<li>Don’t know ground truth, hard to tell what is performing better</li>
<li>Replication failure rate (RFR) among PIPs proposed in <span class="citation" data-cites="cuiImprovingFinemappingModeling2023">(<a href="#/references" role="doc-biblioref" onclick="">10</a>)</span> may support claim that using GIBSS &gt; Linear + RSS &gt; Logistic + RSS</li>
</ul>
<div class="fragment">
<ul>
<li>Other ideas?</li>
<li>Do you know of imbalanced case-control GWAS, survival GWAS, etc. GWAS on count based phenotypes, etc to test out?</li>
</ul>
</div>
</section>
<section id="simulation" class="slide level2 smaller scrollable">
<h2>Simulation</h2>
<p><span class="math display">\[
\begin{align*}
y_i &amp;\sim Bin\left(1, \sigma \left(b_0 + \sum_{j=1}^q b_j x_{ij} + \delta\right)\right)\\
b &amp;\sim N(0, \sigma^2) \\
\delta &amp;\sim N(0, \nu - q \sigma^2)\\
\end{align*}
\]</span></p>
<table>
<colgroup>
<col style="width: 50%">
<col style="width: 50%">
</colgroup>
<thead>
<tr class="header">
<th>Value</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(X\)</span></td>
<td>Standardized genotypes</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\sigma^2\)</span></td>
<td>Variance of standardized effects, i.e.&nbsp;<span class="math inline">\(b \sim N(0, \sigma^2)\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(q\)</span></td>
<td>Number of causal variants in locus</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\rho\)</span></td>
<td>Fraction of variance of genetic component in-locus</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(k\)</span></td>
<td>Fraction of cases (determines <span class="math inline">\(b_0\)</span>)</td>
</tr>
<tr class="even">
<td><span class="math inline">\(q\sigma^2\)</span></td>
<td>(Expected) variance of genetic component in-locus</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\nu\)</span></td>
<td><span class="math inline">\(q \sigma^2/\rho\)</span>, (expected) variance of genetic component</td>
</tr>
<tr class="even">
<td><span class="math inline">\(h^2\)</span></td>
<td><span class="math inline">\(\nu / (\nu + \pi^2/3)\)</span>, (expected) heritability of liability <sup>1</sup></td>
</tr>
</tbody>
</table>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="resources/falconer_liability_threshold.png"></p>
<p></p><figcaption>Liability threshold model, borrowed from <span class="citation" data-cites="falconerInheritanceLiabilityCertain1965">(<a href="#/references" role="doc-biblioref" onclick="">11</a>)</span></figcaption><p></p>
</figure>
</div>
<aside><ol class="aside-footnotes"><li id="fn3"><p>see <span class="citation" data-cites="falconerInheritanceLiabilityCertain1965">(<a href="#/references" role="doc-biblioref" onclick="">11</a>)</span>, and make sure this makes sense!</p></li></ol></aside></section>
<section id="examples-where-susie-is-applied-to-non-gaussian-linear-summary-stats" class="slide level2">
<h2>Examples where SuSiE is applied to non-Gaussian linear summary stats</h2>
<p><span class="citation" data-cites="wightmanGenomewideAssociationStudy2021">(<a href="#/references" role="doc-biblioref" onclick="">8</a>)</span>, Alzheimers meta analysis combining linear and logistic association studies <span class="citation" data-cites="kurkiFinnGenProvidesGenetic2023">(<a href="#/references" role="doc-biblioref" onclick="">9</a>)</span> logistic-mixed model SAIGE + SuSiE</p>
</section>
<section id="when-does-logistic-wakefield-perform-poorly" class="slide level2 smaller scrollable">
<h2>When does Logistic + Wakefield perform poorly?</h2>
<p><img data-src="resources/q3_fdp.png"></p>
<p><img data-src="resources/q3_table.png"></p>
<ul>
<li>accidentally wiped the simulations, need to regenerate</li>
<li>apparently</li>
</ul>
</section>
<section id="dealing-with-intercept-covariates" class="slide level2">
<h2>Dealing with intercept + covariates</h2>
<p>A few options:</p>
<ol type="1">
<li>Estimate in outer loop, treat as a fixed offset while estimating SERs</li>
<li>Re-estimate covariate effects for each variable</li>
<li>Found that reestimating intercept was helpful</li>
</ol>
</section>
<section id="univariate-bf" class="slide level2">
<h2>Univariate BF</h2>

<img data-src="resources/bf_uni.png" class="r-stretch quarto-figure-center"><p class="caption">Limiting BF</p></section>
<section id="limiting-bf" class="slide level2">
<h2>Limiting BF</h2>
<p>Idea put a normal prior on all covariates <span class="math inline">\(\begin{bmatrix} \alpha \\ \beta \end{bmatrix} \sim N(\begin{bmatrix} 0 \\ 0 \end{bmatrix}, \begin{bmatrix} I_{p-1} \tau_0^{-1} &amp; 0 \\ 0 &amp; \tau_1^{-1} \end{bmatrix}\)</span> and compute Laplace approximation to the BF. Take <span class="math inline">\(\tau_0 \rightarrow 0+\)</span>.</p>

<img data-src="resources/limiting_bf.png" class="r-stretch"><p><strong>Q: How variable is the scaling factor? Can we get away with just using the univariate BF?</strong></p>
</section>
<section id="better-quadrature-rule" class="slide level2 smaller">
<h2>Better quadrature rule</h2>
<p><strong>Gauss-Hermite quadrature</strong></p>
<p><span class="math display">\[
I = \int f(x) e^{-x^2} dx \approx \sum_{i=1}^n w_i f(x_i)
\]</span></p>
<p><span class="math inline">\((x_i)_{i=1}^n\)</span> are the roots of the Hermite polynomial <span class="math inline">\(H_n(x)\)</span>, <span class="math inline">\(w_i = \frac{2^{n-1} n! \sqrt{\pi}}{n^2 H_{n-1}^2 (x_i)}\)</span></p>
<div class="fragment">
<p><span class="math display">\[
I = \int f(x) dx = \int \left[\frac{f(x)}{q(x)} \right] q(x) dx, \;\; q(x) = N(x | \mu, \sigma^2)\;  \text{s.t.}\;  \frac{f}{q} \approx 1
\]</span></p>
<ul>
<li>Asymptotically correct</li>
<li>Otherwise, integrating a function with little variation where the integrand has mass</li>
<li>Upshot: very accurate integrals with e.g.&nbsp;<span class="math inline">\(n = 8, 16\)</span>.</li>
</ul>
<p>(note: change of variable + scaling factor to apply the <span class="math inline">\(n\)</span> point Hermite quadrature rule)</p>
</div>
</section>
<section class="slide level2 smaller scrollable" id="references">


<div class="footer footer-default">

</div>
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-wangSimpleNewApproach" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Wang G, Sarkar A, Carbonetto P, Stephens M. A simple new approach to variable selection in regression, with application to genetic fine-mapping. :41. </div>
</div>
<div id="ref-hastieGeneralizedAdditiveModels1986" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Hastie T, Tibshirani R. Generalized <span>Additive Models</span>. Statistical Science [Internet]. 1986 Aug [cited 2024 Jan 18];1(3):297–310. Available from: <a href="https://projecteuclid.org/journals/statistical-science/volume-1/issue-3/Generalized-Additive-Models/10.1214/ss/1177013604.full">https://projecteuclid.org/journals/statistical-science/volume-1/issue-3/Generalized-Additive-Models/10.1214/ss/1177013604.full</a></div>
</div>
<div id="ref-hastieCloserLookDeviance1987" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Hastie T. A <span>Closer Look</span> at the <span>Deviance</span>. The American Statistician. 1987;Volume 41:16–20. </div>
</div>
<div id="ref-zhurss" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Zhu X, Stephens M. Bayesian large-scale multiple regression with summary statistics from genome-wide association studies. Ann Appl Stat [Internet]. 2017 Sep 1 [cited 2022 Jun 30];11(3). Available from: <a href="https://projecteuclid.org/journals/annals-of-applied-statistics/volume-11/issue-3/Bayesian-large-scale-multiple-regression-with-summary-statistics-from-genome/10.1214/17-AOAS1046.full">https://projecteuclid.org/journals/annals-of-applied-statistics/volume-11/issue-3/Bayesian-large-scale-multiple-regression-with-summary-statistics-from-genome/10.1214/17-AOAS1046.full</a></div>
</div>
<div id="ref-lozanoMultivariateNormalDistribution2017" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Lozano JA, Hormozdiari F, Joo JW(Joanne), Han B, Eskin E. The <span>Multivariate Normal Distribution Framework</span> for <span>Analyzing Association Studies</span> [Internet]. <span>bioRxiv</span>; 2017 [cited 2024 Jan 2]. p. 208199. Available from: <a href="https://www.biorxiv.org/content/10.1101/208199v1">https://www.biorxiv.org/content/10.1101/208199v1</a></div>
</div>
<div id="ref-hormozdiariIdentifyingCausalVariants2014" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Hormozdiari F, Kostem E, Kang EY, Pasaniuc B, Eskin E. Identifying <span>Causal Variants</span> at <span>Loci</span> with <span>Multiple Signals</span> of <span>Association</span>. Genetics [Internet]. 2014 Oct [cited 2020 Dec 8];198(2):497–508. Available from: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4196608/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4196608/</a></div>
</div>
<div id="ref-susierss2022" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Zou Y, Carbonetto P, Wang G, Stephens M. Fine-mapping from summary data with the <span>“<span>Sum</span> of <span>Single Effects</span>”</span> model. PLOS Genetics [Internet]. 2022 Jul 19 [cited 2023 Apr 3];18(7):e1010299. Available from: <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1010299">https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1010299</a></div>
</div>
<div id="ref-wightmanGenomewideAssociationStudy2021" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Wightman DP, Jansen IE, Savage JE, Shadrin AA, Bahrami S, Holland D, et al. A genome-wide association study with 1,126,563 individuals identifies new risk loci for <span>Alzheimer</span>’s disease. Nat Genet [Internet]. 2021 Sep [cited 2024 Jan 2];53(9, 9):1276–82. Available from: <a href="https://www.nature.com/articles/s41588-021-00921-z">https://www.nature.com/articles/s41588-021-00921-z</a></div>
</div>
<div id="ref-kurkiFinnGenProvidesGenetic2023" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">Kurki MI, Karjalainen J, Palta P, Sipilä TP, Kristiansson K, Donner KM, et al. <span>FinnGen</span> provides genetic insights from a well-phenotyped isolated population. Nature [Internet]. 2023 Jan [cited 2024 Jan 2];613(7944, 7944):508–18. Available from: <a href="https://www.nature.com/articles/s41586-022-05473-8">https://www.nature.com/articles/s41586-022-05473-8</a></div>
</div>
<div id="ref-cuiImprovingFinemappingModeling2023" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Cui R, Elzur RA, Kanai M, Ulirsch JC, Weissbrod O, Daly MJ, et al. Improving fine-mapping by modeling infinitesimal effects. Nat Genet [Internet]. 2023 Nov 30 [cited 2024 Jan 3];1–8. Available from: <a href="https://www.nature.com/articles/s41588-023-01597-3">https://www.nature.com/articles/s41588-023-01597-3</a></div>
</div>
<div id="ref-falconerInheritanceLiabilityCertain1965" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Falconer DS. The inheritance of liability to certain diseases, estimated from the incidence among relatives. Annals of Human Genetics [Internet]. 1965 Aug [cited 2024 Jan 3];29(1):51–76. Available from: <a href="https://onlinelibrary.wiley.com/doi/10.1111/j.1469-1809.1965.tb00500.x">https://onlinelibrary.wiley.com/doi/10.1111/j.1469-1809.1965.tb00500.x</a></div>
</div>
</div>
</section></section>

    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        target: function(trigger) {
          return trigger.previousElementSibling;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'quarto-reveal',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>