<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="description" content="24 examples from package">

<title>Karl’s Website - Kegg Disease Pathways example</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Karl’s Website</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../about.html">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../research.html">
 <span class="menu-text">Research Notes</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#about" id="toc-about" class="nav-link active" data-scroll-target="#about">About</a></li>
  <li><a href="#sers" id="toc-sers" class="nav-link" data-scroll-target="#sers">SERs</a>
  <ul class="collapse">
  <li><a href="#ora" id="toc-ora" class="nav-link" data-scroll-target="#ora">ORA</a></li>
  <li><a href="#ibss" id="toc-ibss" class="nav-link" data-scroll-target="#ibss">IBSS</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Kegg Disease Pathways example</h1>
</div>

<div>
  <div class="description">
    24 examples from package
  </div>
</div>


<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="about" class="level3">
<h3 class="anchored" data-anchor-id="about">About</h3>
<p>This data is the benchmarking data used by Adi Tarca and colleagues in there 2013 PLOS ONE paper benchmarking GSEA methods. The data consist of 24 micro array experiments comparing disease and control samples in relevant tissues. Importantly, each experiment involves a disease with a disease pathway in KEGG. The idea is that good enrichment methods should identify the “target pathway” in KEGG.</p>
<p>Their paper promotes the use of their method PADOG which they show increases the rank of the target pathway compared to other KEGG pathways. Here, we fit logistic SuSiE using the KEGG pathway and gene lists derived the differential expression analysis of each data set (genes with BH corrected <span class="math inline">\(p\)</span>-values <span class="math inline">\(&lt; 0.05\)</span> were included in the gene list).</p>
<p>Overall, we find that most of the differentially expressed genes in each experiment do not show strong evidence for enrichment in the target pathways jointly or marginally. Only one experiment indicated strong evidence for marginal enrichment of the target pathway, and another one was suggestive.</p>
<p>In no case was the target pathway identified in a credible set. Tarca et. al.&nbsp;make a distinction between “metabolic-pathways” and “disease-pathways”. Since all of the target pathways are disease pathways, we may get better results by restricting to the disease pathways alone. Note, that the background gene list will change as a function of the collection of gene sets we use, which could also change the marginal enrichments. It is concerning that not many of the target pathways are even marginally enriched in the current analysis. If the results change dramatically by changing the background gene list/ subsetting the gene sets, I would be concerned by how sensitive the analysis is to the choice of gene sets.</p>
<div class="cell" data-hash="index_cache/html/install_e000ecdb9428e50bb05a1cfec5b32ac2">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># if (!require("BiocManager", quietly = TRUE))</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     install.packages("BiocManager")</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("KEGGdzPathwaysGEO")</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("KEGGandMetacoreDzPathwaysGEO")</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("hgu133plus2.db")  # annotation containing ENTREZ IDs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="index_cache/html/setup_45cf4413e8e90a0065569e3dc3f56258">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(KEGGdzPathwaysGEO)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hgu133plus2.db)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hgu133a.db)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>cache_dir <span class="ot">&lt;-</span> <span class="st">'~/Documents/karltayeb_website/research/kegg_dz_pathway_example/cache/'</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>cache_rds <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(xfun<span class="sc">::</span>cache_rds, <span class="at">dir =</span> cache_dir, <span class="at">clean=</span>T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="index_cache/html/de-funs_fe12a0e78fd42b48fd94c608a596bf10">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>keggdz  <span class="ot">&lt;-</span> <span class="fu">data</span>(<span class="at">package=</span><span class="st">"KEGGdzPathwaysGEO"</span>)<span class="sc">$</span>results[,<span class="st">"Item"</span>]</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>run_de <span class="ot">&lt;-</span> <span class="cf">function</span>(dataset_name, <span class="at">package=</span><span class="st">'KEGGdzPathwaysGEO'</span>){</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># load data into `dat` object</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data</span>(<span class="at">list=</span>dataset_name)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  dat <span class="ot">&lt;-</span> <span class="fu">get</span>(dataset_name)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get map for probeset to ENTREZID</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># looks for annotationENTREZID, from annotation.db package</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">annotation</span>(dat)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  db_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(x, <span class="st">'ENTREZID'</span>)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  db <span class="ot">&lt;-</span> <span class="fu">get</span>(db_name)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  ids <span class="ot">&lt;-</span> <span class="fu">featureNames</span>(dat)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  ids2entrez <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">unlist</span>(<span class="fu">as.list</span>(db[ids])))</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(ids2entrez) <span class="ot">&lt;-</span> <span class="st">'ENTREZ'</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>  ids2entrez<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ids2entrez)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># DE fit with limma</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  pdat <span class="ot">&lt;-</span> <span class="fu">pData</span>(dat)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>  group <span class="ot">&lt;-</span> pdat<span class="sc">$</span>Group</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>  block <span class="ot">&lt;-</span> pdat<span class="sc">$</span>Block <span class="co"># null if no block strcuture</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>  esetm <span class="ot">&lt;-</span> <span class="fu">exprs</span>(dat)</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># two different fits for block and no block design</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># code copied from Adi Tarca, PLOS ONE 2013 comparison paper</span></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(block)){</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>    G <span class="ot">&lt;-</span> <span class="fu">factor</span>(group)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>    design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span><span class="sc">+</span>G)</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">levels</span>(G)</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(esetm, design)</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>    cont.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts=</span><span class="st">"d-c"</span>,<span class="at">levels=</span>design)</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>    fit2 <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit, cont.matrix)</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>    fit2 <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit2)</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>    aT1<span class="ot">&lt;-</span><span class="fu">topTable</span>(fit2,<span class="at">coef=</span><span class="dv">1</span>, <span class="at">number=</span><span class="fu">dim</span>(esetm)[<span class="dv">1</span>])</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>  }<span class="cf">else</span>{</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>    G <span class="ot">&lt;-</span> <span class="fu">factor</span>(group)</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>    block <span class="ot">&lt;-</span> <span class="fu">factor</span>(block)</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>    design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span><span class="sc">+</span>G<span class="sc">+</span>block)</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(design)<span class="ot">&lt;-</span><span class="fu">substr</span>(<span class="fu">colnames</span>(design),<span class="dv">2</span>,<span class="dv">100</span>)</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(esetm, design)</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>    cont.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts=</span><span class="st">"d-c"</span>,<span class="at">levels=</span>design)</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>    fit2 <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit, cont.matrix)</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>    fit2 <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit2)</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>    aT1<span class="ot">&lt;-</span><span class="fu">topTable</span>(fit2,<span class="at">coef=</span><span class="dv">1</span>, <span class="at">number=</span><span class="fu">dim</span>(esetm)[<span class="dv">1</span>])</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get results</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> aT1 <span class="sc">%&gt;%</span> </span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">ID =</span> <span class="fu">rownames</span>(aT1)) <span class="sc">%&gt;%</span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">right_join</span>(ids2entrez) <span class="sc">%&gt;%</span> <span class="co"># map entrezids</span></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(ENTREZ) <span class="sc">%&gt;%</span></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(AveExpr <span class="sc">==</span> <span class="fu">max</span>(AveExpr)) <span class="co"># resolve duplicate probesets</span></span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="index_cache/html/run-de_7796ae4d26652fb464cb5d54c5c0c713">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run and cache de analysis for 24 datasets</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>de <span class="ot">&lt;-</span> <span class="fu">cache_rds</span>(purrr<span class="sc">::</span><span class="fu">map</span>(keggdz, <span class="sc">~</span><span class="fu">run_de</span>(.x)), <span class="at">file=</span><span class="st">'de_data'</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(de) <span class="ot">&lt;-</span> keggdz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="index_cache/html/load-pathways_92ee25149553c7343270d30eae14a3f0">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load gene sets</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> gseasusie<span class="sc">:::</span><span class="fu">load.webGestalt.geneSet</span>(<span class="st">'pathway_KEGG'</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>kegg2 <span class="ot">&lt;-</span> gseasusie<span class="sc">:::</span><span class="fu">convertGeneSet</span>(kegg<span class="sc">$</span>geneSet, <span class="at">min.size =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Adding missing grouping variables: `geneSet`</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># matched gene sets for each data set, Table 1 in Tarca et al PLOS ONE paper</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>target_pathways <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span>dataset, <span class="sc">~</span>target_pathway,</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE1297"</span>, <span class="st">"hsa05010"</span>,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE5281"</span>, <span class="st">"hsa05010"</span>,</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE5281"</span>, <span class="st">"hsa05010"</span>,</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE5281"</span>, <span class="st">"hsa05010"</span>,</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE20153"</span>, <span class="st">"hsa05012"</span>,</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE20291"</span>, <span class="st">"hsa05012"</span>,</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE8762"</span>, <span class="st">"hsa05016"</span>,</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE4107"</span>, <span class="st">"hsa05210"</span>,</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE8671"</span>, <span class="st">"hsa05210"</span>,</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE9348"</span>, <span class="st">"hsa05210"</span>,</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE14762"</span>, <span class="st">"hsa05211"</span>,</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE781"</span>, <span class="st">"hsa05211"</span>,</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE15471"</span>, <span class="st">"hsa05212"</span>,</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE16515"</span>, <span class="st">"hsa05212"</span>,</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE19728"</span>, <span class="st">"hsa05214"</span>,</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE21354"</span>, <span class="st">"hsa05214"</span>,</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE6956"</span>, <span class="st">"hsa05215"</span>,</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE6956"</span>, <span class="st">"hsa05215"</span>,</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE3467"</span>, <span class="st">"hsa05216"</span>,</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE3678"</span>, <span class="st">"hsa05216"</span>,</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE9476"</span>, <span class="st">"hsa05221"</span>,</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE18842"</span>, <span class="st">"hsa05223"</span>,</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE19188"</span>, <span class="st">"hsa05223"</span>,</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GSE3585"</span>, <span class="st">"hsa05414"</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="at">by=</span><span class="fu">join_by</span>(target_pathway <span class="sc">==</span> geneSet), kegg<span class="sc">$</span>geneSetDes)</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(target_pathways)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 3
  dataset  target_pathway description      
  &lt;chr&gt;    &lt;chr&gt;          &lt;chr&gt;            
1 GSE1297  hsa05010       Alzheimer disease
2 GSE5281  hsa05010       Alzheimer disease
3 GSE5281  hsa05010       Alzheimer disease
4 GSE5281  hsa05010       Alzheimer disease
5 GSE20153 hsa05012       Parkinson disease
6 GSE20291 hsa05012       Parkinson disease</code></pre>
</div>
</div>
</section>
<section id="sers" class="level2">
<h2 class="anchored" data-anchor-id="sers">SERs</h2>
<div class="cell" data-hash="index_cache/html/fit-funs_d8092ff3d17c4483bebb51309a591ca5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>get_regression_data <span class="ot">&lt;-</span> <span class="cf">function</span>(study, genesets, threshold){</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  study_background <span class="ot">&lt;-</span> study<span class="sc">$</span>ENTREZ</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  gene_list <span class="ot">&lt;-</span> study_background[<span class="fu">which</span>(study<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> threshold)]</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> gseasusie<span class="sc">:::</span><span class="fu">geneSet2X</span>(genesets)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter down to genes in at least one gene set</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  de_sub <span class="ot">&lt;-</span> study <span class="sc">%&gt;%</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(ENTREZ <span class="sc">%in%</span> <span class="fu">rownames</span>(X)) <span class="sc">%&gt;%</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">adj.P.Val =</span> <span class="fu">p.adjust</span>(P.Value, <span class="at">method=</span><span class="st">'BH'</span>))</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter down gene set matrix</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>  X_sub <span class="ot">&lt;-</span> X[de_sub<span class="sc">$</span>ENTREZ,]</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  X_sub <span class="ot">&lt;-</span> X_sub[, <span class="fu">colSums</span>(X_sub) <span class="sc">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(de_sub<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> threshold)</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">X=</span>X_sub, <span class="at">y=</span>y, <span class="at">dataset =</span> study<span class="sc">$</span>dataset[<span class="dv">1</span>]))</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>fit_ser_kegg <span class="ot">&lt;-</span> <span class="cf">function</span>(dataset, thresh) {</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(dataset)</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pull out de results for this dataset</span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  study <span class="ot">&lt;-</span> de[[dataset]]</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>  study<span class="sc">$</span>dataset <span class="ot">&lt;-</span> dataset</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make regression data</span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>  dat <span class="ot">&lt;-</span> <span class="fu">get_regression_data</span>(study, kegg2, <span class="at">threshold =</span> thresh)</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># fit ser</span></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>  ser <span class="ot">&lt;-</span> <span class="fu">with</span>(dat, logisticsusie<span class="sc">::</span><span class="fu">fit_glm_ser2</span>(</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>    X, y, <span class="at">estimate_prior_variance=</span>T))</span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>  ser<span class="sc">$</span>cs <span class="ot">&lt;-</span> logisticsusie<span class="sc">::</span><span class="fu">compute_cs</span>(ser<span class="sc">$</span>alpha)</span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make summary table</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>  study_id <span class="ot">&lt;-</span> dat<span class="sc">$</span>dataset</span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>  target_pathway <span class="ot">&lt;-</span> <span class="fu">with</span>(target_pathways, target_pathway[</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">which</span>(dataset <span class="sc">==</span> study_id)])</span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>  gs_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dat<span class="sc">$</span>X)</span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>  ser_tbl <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathway=</span>gs_names,</span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">lbf =</span> ser<span class="sc">$</span>lbf,</span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha=</span>ser<span class="sc">$</span>alpha, <span class="at">post_mean =</span> ser<span class="sc">$</span>mu, <span class="at">post_var =</span> ser<span class="sc">$</span>var,</span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">betahat =</span> ser<span class="sc">$</span>betahat, <span class="at">shat2 =</span> ser<span class="sc">$</span>shat2, <span class="at">intercept =</span> ser<span class="sc">$</span>intercept,</span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">lr =</span> ser<span class="sc">$</span>lr) <span class="sc">%&gt;%</span></span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(lbf)) <span class="sc">%&gt;%</span></span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>      <span class="at">dataset =</span> study_id,</span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>      <span class="at">rank =</span> <span class="fu">row_number</span>()</span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(ser_tbl)</span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a>fit_ibss_kegg <span class="ot">&lt;-</span> <span class="cf">function</span>(dataset, <span class="at">L=</span><span class="dv">5</span>, <span class="at">thresh=</span><span class="fl">0.05</span>, <span class="at">maxit=</span><span class="dv">10</span>, ...) {</span>
<span id="cb9-56"><a href="#cb9-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(dataset)</span>
<span id="cb9-57"><a href="#cb9-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-58"><a href="#cb9-58" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pull out de results for this dataset</span></span>
<span id="cb9-59"><a href="#cb9-59" aria-hidden="true" tabindex="-1"></a>  study <span class="ot">&lt;-</span> de[[dataset]]</span>
<span id="cb9-60"><a href="#cb9-60" aria-hidden="true" tabindex="-1"></a>  study<span class="sc">$</span>dataset <span class="ot">&lt;-</span> dataset</span>
<span id="cb9-61"><a href="#cb9-61" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-62"><a href="#cb9-62" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make regression data</span></span>
<span id="cb9-63"><a href="#cb9-63" aria-hidden="true" tabindex="-1"></a>  dat <span class="ot">&lt;-</span> <span class="fu">get_regression_data</span>(study, kegg2, <span class="at">threshold =</span> thresh)</span>
<span id="cb9-64"><a href="#cb9-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-65"><a href="#cb9-65" aria-hidden="true" tabindex="-1"></a>  <span class="co"># fit logistic SuSiE</span></span>
<span id="cb9-66"><a href="#cb9-66" aria-hidden="true" tabindex="-1"></a>  tictoc<span class="sc">::</span><span class="fu">tic</span>()</span>
<span id="cb9-67"><a href="#cb9-67" aria-hidden="true" tabindex="-1"></a>  ibss <span class="ot">&lt;-</span> <span class="fu">with</span>(dat, logisticsusie<span class="sc">::</span><span class="fu">ibss_from_ser</span>(</span>
<span id="cb9-68"><a href="#cb9-68" aria-hidden="true" tabindex="-1"></a>    X, y, <span class="at">L=</span>L,</span>
<span id="cb9-69"><a href="#cb9-69" aria-hidden="true" tabindex="-1"></a>    <span class="at">ser_function =</span> logisticsusie<span class="sc">::</span>fit_glm_ser2, <span class="at">maxit =</span> maxit, ...))</span>
<span id="cb9-70"><a href="#cb9-70" aria-hidden="true" tabindex="-1"></a>  ibss<span class="sc">$</span>cs <span class="ot">&lt;-</span> logisticsusie<span class="sc">::</span><span class="fu">compute_cs</span>(ibss<span class="sc">$</span>alpha)</span>
<span id="cb9-71"><a href="#cb9-71" aria-hidden="true" tabindex="-1"></a>  ibss<span class="sc">$</span>gene_sets <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dat<span class="sc">$</span>X)</span>
<span id="cb9-72"><a href="#cb9-72" aria-hidden="true" tabindex="-1"></a>  tictoc<span class="sc">::</span><span class="fu">toc</span>()</span>
<span id="cb9-73"><a href="#cb9-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-74"><a href="#cb9-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(ibss)</span>
<span id="cb9-75"><a href="#cb9-75" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-76"><a href="#cb9-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-77"><a href="#cb9-77" aria-hidden="true" tabindex="-1"></a>fit_ora_kegg <span class="ot">&lt;-</span> <span class="cf">function</span>(dataset, thresh, <span class="at">maxit=</span><span class="dv">10</span>, ...) {</span>
<span id="cb9-78"><a href="#cb9-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(dataset)</span>
<span id="cb9-79"><a href="#cb9-79" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-80"><a href="#cb9-80" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pull out de results for this dataset</span></span>
<span id="cb9-81"><a href="#cb9-81" aria-hidden="true" tabindex="-1"></a>  study <span class="ot">&lt;-</span> de[[dataset]]</span>
<span id="cb9-82"><a href="#cb9-82" aria-hidden="true" tabindex="-1"></a>  study<span class="sc">$</span>dataset <span class="ot">&lt;-</span> dataset</span>
<span id="cb9-83"><a href="#cb9-83" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-84"><a href="#cb9-84" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make regression data</span></span>
<span id="cb9-85"><a href="#cb9-85" aria-hidden="true" tabindex="-1"></a>  dat <span class="ot">&lt;-</span> <span class="fu">get_regression_data</span>(study, kegg2, <span class="at">threshold =</span> thresh)</span>
<span id="cb9-86"><a href="#cb9-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-87"><a href="#cb9-87" aria-hidden="true" tabindex="-1"></a>  <span class="co"># fit logistic SuSiE</span></span>
<span id="cb9-88"><a href="#cb9-88" aria-hidden="true" tabindex="-1"></a>  ora <span class="ot">&lt;-</span> <span class="fu">with</span>(dat, gseasusie<span class="sc">::</span><span class="fu">fit_ora</span>(X, y))</span>
<span id="cb9-89"><a href="#cb9-89" aria-hidden="true" tabindex="-1"></a>  ora<span class="sc">$</span>dataset <span class="ot">&lt;-</span> dataset</span>
<span id="cb9-90"><a href="#cb9-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(ora)</span>
<span id="cb9-91"><a href="#cb9-91" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="index_cache/html/fit_sers_3fb11ac24e7b44df124621c812de419e">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>ser_tbl <span class="ot">&lt;-</span> <span class="fu">cache_rds</span>(purrr<span class="sc">::</span><span class="fu">map</span>(keggdz, <span class="sc">~</span><span class="fu">fit_ser_kegg</span>(.x, <span class="fl">0.05</span>)), <span class="at">file=</span><span class="st">'kegg_ser_fits_bh_05'</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ser_tbl) <span class="ot">&lt;-</span> keggdz</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>ser_res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(ser_tbl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Only 1/24 of the “target gene sets” have strong marginal enrichment, and it is not even the strongest marginal enrichment. Only 6 have positive BFs. That does not bode well for selecting the target gene set with logistic SuSiE.</p>
<div class="cell" data-hash="index_cache/html/target_ranks_d49e3fd935beb2ff979daa7496652808">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># report rank of target pathawy, etc. in each SER</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>ser_res <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(target_pathways) <span class="sc">%&gt;%</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">select</span>(<span class="fu">mutate</span>(kegg<span class="sc">$</span>geneSetDes, <span class="at">pathway=</span>geneSet), <span class="sc">-</span>geneSet)) <span class="sc">%&gt;%</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pathway <span class="sc">==</span> target_pathway) <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pathway, description, dataset, lbf, rank) <span class="sc">%&gt;%</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(lbf))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(dataset)`
Joining with `by = join_by(pathway, description)`</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    pathway                  description  dataset         lbf rank
1  hsa05010            Alzheimer disease  GSE1297 18.87247938    3
2  hsa05214                       Glioma GSE21354  2.01288013   11
3  hsa05214                       Glioma GSE19728  0.42306584   58
4  hsa05211         Renal cell carcinoma GSE14762  0.41946809   73
5  hsa05210            Colorectal cancer  GSE8671  0.23743965   53
6  hsa05216               Thyroid cancer  GSE3678  0.23717420   78
7  hsa05012            Parkinson disease GSE20153 -0.01594011  140
8  hsa05223   Non-small cell lung cancer GSE19188 -0.23591103  231
9  hsa05223   Non-small cell lung cancer GSE18842 -0.35737532   68
10 hsa05210            Colorectal cancer  GSE4107 -0.66736583  255
11 hsa05016           Huntington disease  GSE8762 -0.66789614  309
12 hsa05210            Colorectal cancer  GSE9348 -0.72299687  137
13 hsa05216               Thyroid cancer  GSE3467 -0.74701199  139
14 hsa05221       Acute myeloid leukemia  GSE9476 -0.93016715  263
15 hsa05414 Dilated cardiomyopathy (DCM)  GSE3585 -1.02153866  273
16 hsa05012            Parkinson disease GSE20291 -1.05382980  252
17 hsa05212            Pancreatic cancer GSE16515 -1.42235641  217
18 hsa05212            Pancreatic cancer GSE15471 -1.48947468  190
19 hsa05211         Renal cell carcinoma   GSE781 -1.68321790  265</code></pre>
</div>
</div>
<p>Next we look at the top marginally enriched pathway. The strongest marginal enrichments are overwhelmingly metabolic or signalling pathways, rather than the disease pathways.</p>
<div class="cell" data-hash="index_cache/html/top_marginal_enrichment_ser_55edc55198360d2dd266a158ab697f8e">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># report top marginal enrichment in each dataset</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>ser_res <span class="sc">%&gt;%</span> </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(target_pathways) <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(lbf <span class="sc">==</span> <span class="fu">max</span>(lbf)) <span class="sc">%&gt;%</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">target_description =</span> description) <span class="sc">%&gt;%</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(kegg<span class="sc">$</span>geneSetDes, <span class="fu">join_by</span>(pathway <span class="sc">==</span> geneSet)) <span class="sc">%&gt;%</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(dataset, pathway, description, target_pathway, target_description, lbf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(dataset)`</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 24 × 6
# Groups:   dataset [24]
   dataset  pathway  description         target_pathway target_description   lbf
   &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;               &lt;chr&gt;          &lt;chr&gt;              &lt;dbl&gt;
 1 GSE1297  hsa00190 Oxidative phosphor… hsa05010       Alzheimer disease  26.1 
 2 GSE14762 hsa04080 Neuroactive ligand… hsa05211       Renal cell carcin… 23.9 
 3 GSE15471 hsa04740 Olfactory transduc… hsa05212       Pancreatic cancer  16.9 
 4 GSE16515 hsa03050 Proteasome          hsa05212       Pancreatic cancer   8.92
 5 GSE18842 hsa04740 Olfactory transduc… hsa05223       Non-small cell lu… 41.1 
 6 GSE19188 hsa00533 Glycosaminoglycan … hsa05223       Non-small cell lu…  3.88
 7 GSE19728 hsa04060 Cytokine-cytokine … hsa05214       Glioma             10.6 
 8 GSE20153 hsa04141 Protein processing… hsa05012       Parkinson disease   4.87
 9 GSE20291 hsa04978 Mineral absorption  hsa05012       Parkinson disease   8.24
10 GSE21354 hsa04060 Cytokine-cytokine … hsa05214       Glioma              6.01
# ℹ 14 more rows</code></pre>
</div>
</div>
<p>Below we summarize the evidence for <em>an</em> effect. The SER model with the strongest evidence for <em>any</em> enrichment has a <span class="math inline">\(BF \approx 120\)</span>, The smalleset has <span class="math inline">\(BF \approx 1.3\)</span>. That is the ratio of the marginal likelihood for the <em>single effect model</em> compared to the null model. Even though many of the microarray datasets do not show strong enerichment in their target pathway, many (~15/24) do show moderate to strong evidence for enrichment in <em>a</em> pathway.</p>
<div class="cell" data-hash="index_cache/html/ser_cs_bec50bf63480c9c5e0de83966c19df1e">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>global_evidence <span class="ot">&lt;-</span> ser_res <span class="sc">%&gt;%</span> </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">log_bf_ser =</span> matrixStats<span class="sc">::</span><span class="fu">logSumExp</span>(lbf) <span class="sc">-</span> <span class="fu">log</span>(<span class="fu">length</span>(lbf)),</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">cs =</span> <span class="fu">list</span>(logisticsusie<span class="sc">::</span><span class="fu">compute_cs</span>(alpha)),</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">max_pip =</span> <span class="fu">max</span>(alpha)</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(log_bf_ser)) <span class="sc">%&gt;%</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cs_size =</span> cs<span class="sc">$</span>size)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>global_evidence</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 24 × 5
# Rowwise: 
   dataset     log_bf_ser cs     max_pip cs_size
   &lt;chr&gt;            &lt;dbl&gt; &lt;list&gt;   &lt;dbl&gt;   &lt;dbl&gt;
 1 GSE8671          40.6  &lt;cs&gt;     1.00        1
 2 GSE781           36.4  &lt;cs&gt;     1.00        1
 3 GSE18842         35.3  &lt;cs&gt;     1.00        1
 4 GSE5281_HIP      26.4  &lt;cs&gt;     1.00        1
 5 GSE1297          20.3  &lt;cs&gt;     0.998       1
 6 GSE14762         18.2  &lt;cs&gt;     0.943       2
 7 GSE4107          15.5  &lt;cs&gt;     0.999       1
 8 GSE15471         11.3  &lt;cs&gt;     0.839       2
 9 GSE9348           9.16 &lt;cs&gt;     0.846       2
10 GSE3467           5.71 &lt;cs&gt;     0.977       1
# ℹ 14 more rows</code></pre>
</div>
</div>
<section id="ora" class="level3">
<h3 class="anchored" data-anchor-id="ora">ORA</h3>
<p>We also check the ranks of the marginal enrichments when we do the standard ORA (that is, Fisher’s exact test for 2x2 contingency table). The ranks differ a bit compared to the logistic regression but are broadly similar.</p>
<div class="cell" data-hash="index_cache/html/fit_ora_and_ibss_5473a5ec9dc312c3608dd03dd9dd8bb8">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>ora_res <span class="ot">&lt;-</span> <span class="fu">cache_rds</span>(purrr<span class="sc">::</span><span class="fu">map</span>(keggdz, <span class="sc">~</span><span class="fu">fit_ora_kegg</span>(.x, <span class="fl">0.05</span>)), <span class="at">file=</span><span class="st">'kegg_ora_bh_05'</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>ora_res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(ora_res)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="index_cache/html/unnamed-chunk-1_8a6e9c2c55ab898707558b9e32a4f1b6">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>ora_res <span class="sc">%&gt;%</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(target_pathways) <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(kegg<span class="sc">$</span>geneSetDes) <span class="sc">%&gt;%</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(pFishersExact) <span class="sc">%&gt;%</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">row_number</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(geneSet <span class="sc">==</span> target_pathway) <span class="sc">%&gt;%</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(geneSet, dataset, target_pathway, description, pFishersExact, rank)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(dataset)`
Joining with `by = join_by(geneSet, description)`</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 19 × 6
# Groups:   dataset [19]
   geneSet  dataset  target_pathway description              pFishersExact  rank
   &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;          &lt;chr&gt;                            &lt;dbl&gt; &lt;int&gt;
 1 hsa05010 GSE1297  hsa05010       Alzheimer disease             6.19e-11     3
 2 hsa05214 GSE21354 hsa05214       Glioma                        8.82e- 3    11
 3 hsa05210 GSE8671  hsa05210       Colorectal cancer             6.41e- 2    54
 4 hsa05214 GSE19728 hsa05214       Glioma                        6.44e- 2    54
 5 hsa05211 GSE14762 hsa05211       Renal cell carcinoma          6.63e- 2    74
 6 hsa05216 GSE3678  hsa05216       Thyroid cancer                9.40e- 2    80
 7 hsa05223 GSE18842 hsa05223       Non-small cell lung can…      1.08e- 1    72
 8 hsa05210 GSE9348  hsa05210       Colorectal cancer             2.24e- 1   106
 9 hsa05216 GSE3467  hsa05216       Thyroid cancer                3.18e- 1   133
10 hsa05016 GSE8762  hsa05016       Huntington disease            3.97e- 1    94
11 hsa05212 GSE16515 hsa05212       Pancreatic cancer             4.15e- 1   151
12 hsa05012 GSE20153 hsa05012       Parkinson disease             4.41e- 1   105
13 hsa05212 GSE15471 hsa05212       Pancreatic cancer             4.98e- 1   166
14 hsa05012 GSE20291 hsa05012       Parkinson disease             5.55e- 1   212
15 hsa05210 GSE4107  hsa05210       Colorectal cancer             5.81e- 1   215
16 hsa05414 GSE3585  hsa05414       Dilated cardiomyopathy …      6.40e- 1   209
17 hsa05211 GSE781   hsa05211       Renal cell carcinoma          7.53e- 1   268
18 hsa05223 GSE19188 hsa05223       Non-small cell lung can…      8.85e- 1   261
19 hsa05221 GSE9476  hsa05221       Acute myeloid leukemia        8.94e- 1   281</code></pre>
</div>
</div>
</section>
<section id="ibss" class="level3">
<h3 class="anchored" data-anchor-id="ibss">IBSS</h3>
<div class="cell" data-hash="index_cache/html/fit-ibss_7c10ab970b1bb3425de755910377e52b">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># take datasets with BF SER &gt; 10 to fit with IBSS</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>keggdz_global_enrichment <span class="ot">&lt;-</span> global_evidence<span class="sc">$</span>dataset[(global_evidence<span class="sc">$</span>log_bf_ser <span class="sc">&gt;</span> <span class="fu">log</span>(<span class="dv">10</span>))]</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>ibss <span class="ot">&lt;-</span> <span class="fu">cache_rds</span>(</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map</span>(keggdz_global_enrichment,</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>             <span class="sc">~</span><span class="fu">fit_ibss_kegg</span>(.x, <span class="at">L=</span><span class="dv">5</span>, <span class="at">thresh =</span> <span class="fl">0.05</span>, <span class="at">maxit=</span><span class="dv">10</span>)),</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">file=</span><span class="st">'kegg_gibss_fits_bh_05'</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ibss) <span class="ot">&lt;-</span> keggdz_global_enrichment</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(dataset <span class="cf">in</span> keggdz_global_enrichment){</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>  fit <span class="ot">&lt;-</span> ibss[[dataset]]</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>  ibss[[dataset]]<span class="sc">$</span>prior_variance <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_dbl</span>(</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>    fit<span class="sc">$</span>fits, <span class="sc">~</span>.x<span class="sc">$</span>prior_variance)</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>  ibss[[dataset]]<span class="sc">$</span>lbf_ser <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_dbl</span>(</span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>    fit<span class="sc">$</span>fits, <span class="sc">~</span>.x<span class="sc">$</span>lbf_model)</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="index_cache/html/unnamed-chunk-2_e96fc0f8d08dada0e7b1400e99e148b5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>prior_variance_ibss <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_dfr</span>(ibss, <span class="sc">~</span>.x<span class="sc">$</span>prior_variance) <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">dataset =</span> <span class="fu">names</span>(ibss)) <span class="sc">%&gt;%</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(dataset)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>prior_variance_ibss</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 16 × 6
   dataset        L1    L2     L3     L4     L5
   &lt;chr&gt;       &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 GSE8671     1.47  1.99  3.20   1.81   6.95  
 2 GSE781      3.73  4.52  1.37   1.95   0.741 
 3 GSE18842    3.19  0.772 1.77   0.376  0.440 
 4 GSE5281_HIP 2.85  2.11  1.17   3.36   0.826 
 5 GSE1297     2.24  0.100 0.100  0.100  0.100 
 6 GSE14762    0.900 1.89  1.35   1.25   1.21  
 7 GSE4107     0.142 1.31  0.662  0.689  2.09  
 8 GSE15471    3.80  0.952 1.18   0.171  0.171 
 9 GSE9348     1.16  0.460 0.674  0.560  0.232 
10 GSE3467     1.17  0.333 0.571  0.451  0.436 
11 GSE19728    0.546 0.184 0.184  0.185  0.185 
12 GSE6956C    1.03  1.86  0.0438 0.0438 0.0438
13 GSE3678     0.614 1.19  0.198  0.198  0.198 
14 GSE5281_VCX 0.637 0.586 0.763  1.05   0.299 
15 GSE16515    2.19  0.385 0.163  0.163  0.163 
16 GSE20291    3.15  0.474 0.474  0.474  0.475 </code></pre>
</div>
</div>
<div class="cell" data-hash="index_cache/html/unnamed-chunk-3_f9e8a06f6dffd7dc98a2315cc7fc0668">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>bf_ser_ibss <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_dfr</span>(ibss, <span class="sc">~</span>.x<span class="sc">$</span>lbf_ser) <span class="sc">%&gt;%</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">dataset =</span> <span class="fu">names</span>(ibss)) <span class="sc">%&gt;%</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(dataset)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>bf_ser_ibss</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 16 × 6
   dataset        L1    L2    L3    L4    L5
   &lt;chr&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
 1 GSE8671     137.  137.  138.  131.  138. 
 2 GSE781       97.0  97.4  97.4  97.4  99.4
 3 GSE18842     94.7  94.9  95.2  95.2  98.7
 4 GSE5281_HIP 106.  106.  106.  106.  104. 
 5 GSE1297      24.5  31.9  31.9  31.9  31.9
 6 GSE14762     48.6  48.7  60.0  60.2  60.3
 7 GSE4107      72.1  72.2  70.6  71.0  68.2
 8 GSE15471     52.6  49.7  52.2  58.0  58.0
 9 GSE9348      47.5  47.6  47.6  49.1  52.0
10 GSE3467      32.1  32.3  33.1  36.5  36.7
11 GSE19728     18.0  23.2  23.1  23.1  23.1
12 GSE6956C     11.7  16.0  21.9  21.9  21.9
13 GSE3678      24.4  24.7  29.5  29.5  29.5
14 GSE5281_VCX  37.3  37.3  37.4  38.2  41.1
15 GSE16515     21.5  21.3  26.9  26.9  26.9
16 GSE20291     11.3  17.6  17.6  17.6  17.6</code></pre>
</div>
</div>
<section id="gse1297" class="level4">
<h4 class="anchored" data-anchor-id="gse1297">GSE1297</h4>
<p>When we fit logistic SuSiE for <code>GSE1297</code> we see that the first component selects the top marginally enriched gene set. All other components have a small estimated prior variance, very wide credible sets, and small SER BFs. The top marginal enrichment is “Oxidative phosphorylation”, not the target pathway “Alzheimer’s disease”.</p>
<p>If we look at the marginal enrichment results, filtering at BH corrected <span class="math inline">\(p\)</span>-values &lt; 0.05 we find 8 marginal enrichment. “Alzehimer’s disease” gene set is moderately correlated with the top marginal enrichment (which is captured by the <code>L1</code> component) with a correlation of <span class="math inline">\(0.56\)</span>. However, we see that some of the marginally enriched gene sets are only weakly correlated with the top marginal enrichment. We would hope that these are identified by another component in the SuSiE fit, but they are not.</p>
<div class="cell" data-hash="index_cache/html/show_marginal_results_0618a7de7533bee99c7a2f5b6d6c56b5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>marginal_enrichments <span class="ot">&lt;-</span> ora_res <span class="sc">%&gt;%</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(dataset <span class="sc">==</span> <span class="st">'GSE1297'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pAdj =</span> <span class="fu">p.adjust</span>(pFishersExact, <span class="at">method=</span><span class="st">'BH'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pAdj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(kegg<span class="sc">$</span>geneSetDes) <span class="sc">%&gt;%</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(geneSet, description, oddsRatio, pAdj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(geneSet)`</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>marginal_enrichments <span class="sc">%&gt;%</span> </span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>((pAdj))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 × 4
  geneSet  description                               oddsRatio     pAdj
  &lt;chr&gt;    &lt;chr&gt;                                         &lt;dbl&gt;    &lt;dbl&gt;
1 hsa00190 Oxidative phosphorylation                      4.63 1.24e-11
2 hsa05016 Huntington disease                             3.04 6.30e- 9
3 hsa05010 Alzheimer disease                              3.11 6.81e- 9
4 hsa05012 Parkinson disease                              3.54 1.32e- 8
5 hsa04714 Thermogenesis                                  2.35 8.84e- 6
6 hsa04932 Non-alcoholic fatty liver disease (NAFLD)      2.30 5.25e- 4
7 hsa04723 Retrograde endocannabinoid signaling           2.23 2.33e- 3
8 hsa01100 Metabolic pathways                             1.31 3.96e- 2</code></pre>
</div>
</div>
<div class="cell" data-hash="index_cache/html/lr-marginal_4b2245e1fc8b03d43a51fece3dc5cbbf">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>ser_res <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(dataset <span class="sc">==</span> <span class="st">'GSE1297'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(kegg<span class="sc">$</span>geneSetDes, <span class="at">by=</span><span class="fu">join_by</span>(pathway <span class="sc">==</span> geneSet)) <span class="sc">%&gt;%</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(dataset, pathway, description, lbf, rank) <span class="sc">%&gt;%</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(lbf <span class="sc">&gt;</span> <span class="fu">log</span>(<span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>   dataset  pathway                               description       lbf rank
1  GSE1297 hsa00190                 Oxidative phosphorylation 26.078254    1
2  GSE1297 hsa05016                        Huntington disease 19.361949    2
3  GSE1297 hsa05010                         Alzheimer disease 18.872479    3
4  GSE1297 hsa05012                         Parkinson disease 18.050441    4
5  GSE1297 hsa04714                             Thermogenesis 11.538981    5
6  GSE1297 hsa04932 Non-alcoholic fatty liver disease (NAFLD)  7.449590    6
7  GSE1297 hsa04723      Retrograde endocannabinoid signaling  6.040164    7
8  GSE1297 hsa03040                               Spliceosome  3.302544    8
9  GSE1297 hsa03015                 mRNA surveillance pathway  3.288215    9
10 GSE1297 hsa00620                       Pyruvate metabolism  2.543708   10
11 GSE1297 hsa01100                        Metabolic pathways  2.530046   11
12 GSE1297 hsa00471    D-Glutamine and D-glutamate metabolism  2.335098   12</code></pre>
</div>
</div>
<div class="cell" data-hash="index_cache/html/marginal_enrichment_correlations_26a6c84b41bdaf30c7cc953aa8a27a7b">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">get_regression_data</span>(de<span class="sc">$</span>GSE1297, kegg2, <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Unknown or uninitialised column: `dataset`.</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>R <span class="ot">&lt;-</span> <span class="fu">cor</span>(<span class="fu">as.matrix</span>(dat<span class="sc">$</span>X[, marginal_enrichments<span class="sc">$</span>geneSet]))</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the minimum and maximum correlation</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>max_cor <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">abs</span>(R))</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>min_cor <span class="ot">&lt;-</span> <span class="sc">-</span>max_cor</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create color palette</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>my_palette <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span>))(<span class="dv">100</span>)</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>breaks <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span> min_cor, <span class="at">to =</span> max_cor, <span class="at">length.out =</span> <span class="dv">101</span>)</span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>gplots<span class="sc">::</span><span class="fu">heatmap.2</span>(</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>          R,</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>          <span class="at">main =</span> <span class="st">"Correlation Heatmap"</span>, <span class="co"># main title</span></span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>          <span class="at">trace =</span> <span class="st">"none"</span>, <span class="co"># to not draw trace lines on the heatmap</span></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>          <span class="at">col =</span> my_palette, <span class="co"># color scheme from blue-white-red,</span></span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>          <span class="at">breaks =</span> breaks</span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/marginal_enrichment_correlations-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<div class="cell" data-hash="index_cache/html/unnamed-chunk-4_9c61f87fe7d3b94d4e580f453ca8a32e">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>gse1297 <span class="ot">&lt;-</span> ibss<span class="sc">$</span>GSE1297</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>prior_variance <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_dbl</span>(gse1297<span class="sc">$</span>fits, <span class="sc">~</span>.x<span class="sc">$</span>prior_variance)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(prior_variance)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>       L1        L2        L3        L4        L5 
2.2442834 0.1000776 0.1000212 0.1002237 0.1002565 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>lbf_ser <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_dbl</span>(gse1297<span class="sc">$</span>fits, <span class="sc">~</span>.x<span class="sc">$</span>lbf_model)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(lbf_ser)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>      L1       L2       L3       L4       L5 
24.52011 31.89816 31.89977 31.89400 31.89192 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(gse1297<span class="sc">$</span>fits<span class="sc">$</span>L2<span class="sc">$</span>lbf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>